File: src/Streamarr.Core/Channels/Channel.cs
using System;
using Streamarr.Core.Creators;
using Streamarr.Core.Datastore;

namespace Streamarr.Core.Channels
{
    public enum MembershipStatus
    {
        Unknown = 0,
        Active  = 1,
        None    = 2
    }

    public class Channel : ModelBase
    {
        // Relationship
        public int CreatorId { get; set; }

        // Platform identity
        public PlatformType Platform { get; set; }
        public string PlatformId { get; set; } = string.Empty;
        public string PlatformUrl { get; set; } = string.Empty;

        // Display metadata
        public string Title { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public string ThumbnailUrl { get; set; } = string.Empty;

        // Display order (user-defined)
        public int SortOrder { get; set; }

        // State
        public bool Monitored { get; set; }
        public ChannelStatusType Status { get; set; }
        public DateTime? LastInfoSync { get; set; }
        public MembershipStatus MembershipStatus { get; set; }
        public DateTime? LastMembershipCheck { get; set; }

        // Wanted filter — content types
        public bool DownloadVideos { get; set; } = true;
        public bool DownloadShorts { get; set; } = true;
        public bool DownloadVods { get; set; }
        public bool DownloadLive { get; set; } = true;
        public bool DownloadMembers { get; set; }

        // Wanted filter — word filters
        public string WatchedWords { get; set; } = string.Empty;
        public string IgnoredWords { get; set; } = string.Empty;
        public bool WatchedDefeatsIgnored { get; set; } = true;

        // Download mode
        public bool AutoDownload { get; set; }

        // Retention
        public int? RetentionDays { get; set; }
        public bool KeepVideos { get; set; } = true;
        public bool KeepShorts { get; set; } = true;
        public bool KeepVods { get; set; }
        public bool KeepMembers { get; set; }
        public string RetentionKeepWords { get; set; } = string.Empty;

        // Navigation
        public LazyLoaded<Creator> Creator { get; set; }
    }
}

File: frontend/src/typings/Channel.ts
import ModelBase from 'App/ModelBase';

export type ChannelStatusType = 0 | 1 | 2;

interface Channel extends ModelBase {
  creatorId: number;
  platform: string;
  platformId: string;
  platformUrl: string;
  title: string;
  description: string;
  thumbnailUrl: string;
  category: string;
  sortOrder: number;
  monitored: boolean;
  status: ChannelStatusType;
  lastInfoSync: string | null;
  membershipStatus: 'unknown' | 'active' | 'none';
  lastMembershipCheck: string | null;

  // Wanted — content types
  downloadVideos: boolean;
  downloadShorts: boolean;
  downloadVods: boolean;
  downloadLive: boolean;
  downloadMembers: boolean;

  // Wanted — word filters
  watchedWords: string;
  ignoredWords: string;
  watchedDefeatsIgnored: boolean;

  // Download mode
  autoDownload: boolean;

  // Retention
  retentionDays: number | null;
  keepVideos: boolean;
  keepShorts: boolean;
  keepVods: boolean;
  keepMembers: boolean;
  retentionKeepWords: string;
}

export default Channel;
